File: articles/overconfidence-bias-financial-markets/fortran/overconfidence_demand.f90
program overconfidence_demand
  implicit none
  real :: expected_return, perceived_variance, alpha, beta
  real :: demand, gross_return, trading_intensity, cost_per_turnover, net_return

  expected_return = 0.08
  perceived_variance = 0.03
  alpha = 1.2
  beta = 0.7
  demand = alpha * expected_return - beta * perceived_variance

  gross_return = 0.05
  trading_intensity = 1.4
  cost_per_turnover = 0.0025
  net_return = gross_return - cost_per_turnover * trading_intensity

  print *, "Synthetic investor demand:", demand
  print *, "Synthetic net return after cost:", net_return
end program overconfidence_demand

File: articles/overconfidence-bias-financial-markets/rust/overconfidence_demand.rs
fn investor_demand(expected_return: f64, perceived_variance: f64, alpha: f64, beta: f64) -> f64 {
    alpha * expected_return - beta * perceived_variance
}

fn net_return_after_cost(gross_return: f64, trading_intensity: f64, cost_per_turnover: f64) -> f64 {
    gross_return - cost_per_turnover * trading_intensity
}

fn main() {
    println!("Synthetic investor demand: {:.3}", investor_demand(0.08, 0.03, 1.2, 0.7));
    println!("Synthetic net return after cost: {:.3}", net_return_after_cost(0.05, 1.4, 0.0025));
}
